python/dlr/counter/system.py

```diff
@@ -0,0 +1,217 @@
+"""System interface"""
+import platform
+import uuid
+import logging
+import abc
+
+from .deviceinfo import DeviceInfo
+from .utils.helper import get_hash_string
+
+
+# Interface
+class System:
+    """Root Interface of Systems hierarchy"""
+
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def get_device_info(self):
+        """Return a list of device information"""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_device_uuid(self):
+        """Return DeviceInfo uuid"""
+        raise NotImplementedError
+
+
+# Wrapper class
+class ARM(System):
```

Comment:
Is it necessary to have all of these subclasses for each type of system? Is there really so much of a difference between the platform information from python apis between each type of system? It looks like most implementations of `System` are the same.

Can we just always collect all of the available data from `platform` and let the backend handle of the parsing and processing?

Reply:
I agree. This was written with different system support at first. But I have same remark with the original author. Let me revise this and get back to you.